lib/galaxy/datatypes/images.py

```diff
@@ -285,6 +288,36 @@ def sniff_prefix(self, file_prefix):
         return False
 
 
+class Mrc2014(Binary):
+    """
+    MRC/CCP4 2014 file format (.mrc).
+    https://www.ccpem.ac.uk/mrc_format/mrc2014.php
+
+    >>> from galaxy.datatypes.sniff import get_test_fname
+    >>> fname = get_test_fname('1.mrc')
+    >>> Mrc2014().sniff(fname)
+    True
+    >>> fname = get_test_fname('2.txt')
+    >>> Mrc2014().sniff(fname)
+    False
+    """
+    file_ext = 'mrc'
+
+    def sniff(self, filename):
+        # Handle the wierdness of mrcfile:
+        # https://github.com/ccpem/mrcfile/blob/master/mrcfile/validator.py#L88
+        with contextlib.redirect_stdout(StringIO()):
+            try:
+                # An exception is thrown
+                # if the file is not an
+                # mrc2014 file.
+                if mrcfile.validate(filename):
```

Comment:
Sorry about the misunderstanding, the idea is to use pass `StringIO()` to the `print_file` argument of  `mrcfile.validate`, to avoid the global `contextlib.redirect_stdout`

Reply:
Ah, makes sense.  Sorry about this!